dart/common/LocalResource.cpp

```diff
@@ -0,0 +1,139 @@
+#include <cstring>
+#include <iostream>
+#include "dart/common/Console.h"
+#include "LocalResource.h"
+
+namespace dart {
+namespace common {
+
+LocalResource::LocalResource(const std::string& _path)
+  : mFile(std::fopen(_path.c_str(), "rb"))
+{
+  if(!mFile)
+  {
+    dtwarn << "[LocalResource::constructor] Failed opening file '" << _path
+          << "' for reading: " << std::strerror(errno) << "\n";
+  }
+}
+
+LocalResource::~LocalResource()
+{
+  if (!mFile)
+    return;
+
+  if (std::fclose(mFile) == EOF)
+  {
+    dtwarn << "[LocalResource::destructor] Failed closing file.\n";
```

Comment:
Interesting. The [documentation I found](http://www.cplusplus.com/reference/cstdio/fclose/) just says that the function returns `EOF`. It can't hurt to check `errno` and, if it's set, print the `std::strerror(errno)` anyway.